### ISSUE TITLE
Node container

### ISSUE THREAD

**Issue by vcbnm** [Wouterdek/NodeNetwork#24]
I'd like to suggest a container node which allows attachment of nodes. Having many nodes in a single window will make the view look cluttered and creating complex node graphs will result in many lines crossing each other or passing behind other nodes. The container should allow the same collapsing functionality as normal nodes without breaking outward-going connections. Even better would be if attached nodes can have their in/out ports connected to the container, creating ports on the container itself that will redirect the in/output. This would allow for isolated networks within the master network.

**Comment by Wouterdek** [comment]
This is indeed a very useful concept. In my personal applications, I usually create a special node type for this. This 'group' node contains a separate network viewmodel which specifies the internal working of the group node. Users can then double click the group node to open a second editor for the nested network. Inputs/outputs of the group node are specified using specialized nodes in the nested network. Using this system, it's possible to have great flexibility in structuring the nodes while reusing the existing systems.

I might be able to generalize some of this code and include it in the library, but don't expect this any time soon as it will probably be a while before I can get to that.

**Comment by evilC** [comment]
I concur, this would be super-useful.  
What would be really cool would be if you could group select some existing nodes and "collapse" them into one compound node. Even better if you could then duplicate this group for re-use (And ideally add the group to the node list), kind of a "subroutine" if you will.

**Comment by SadE54** [comment]
Yep , it could be very very useful :)

**Comment by Wouterdek** [comment]
Update: I have created an experimental `grouping` branch that has this functionality. There are only a few issues left, for which I am waiting for the next release of DynamicData.

**Comment by evilC** [comment]
Great to see this library still getting love, thank you!

**Comment by vcbnm** [comment]
Will the grouping feature also allow input/output ports which will redirect towards the sub-nodes?

**Comment by Wouterdek** [comment]
Here is a preview of how it works so far. (ignore the layout issues at the end, this is still WIP :) )
![instant](https://user-images.githubusercontent.com/913101/60676135-663f4080-9e7e-11e9-9d95-f872fd642810.gif)

**Comment by vcbnm** [comment]
I was wondering what the status is, are you still planning to implement this feature?

**Comment by Wouterdek** [comment]
> I was wondering what the status is, are you still planning to implement this feature?

My apologies, it's been almost a year. Last time I worked on this, there were some issues. No real roadblocks, mostly just bugs. I prioritized other issues over this one and as a result, this has gathered dust on the git shelf. ([the "grouping" branch](https://github.com/Wouterdek/NodeNetwork/tree/grouping))

I do still want to implement this at some point, however I'm currently working on my master's thesis and my time is very limited so I unfortunately can't promise that it will be done anytime soon. I don't want to publish buggy, half-implemented features.
